apis/Google.Cloud.PubSub.V1/Google.Cloud.PubSub.V1/SubscriberClientImpl.cs

```diff
@@ -29,6 +29,17 @@
 
 namespace Google.Cloud.PubSub.V1;
 
+// The class follows following conventions for the field names:
+// - An "ackDeadline" is the amount of time before which the message must be acknowledged or have its lease extended.
+// - A suffix of "window" indicates the duration of time before the ackDeadline at which the lease is extended.
+// - A suffix of "interval" indicates the delay between sending one auto-lease extension request and the next; This is derived from the ackDeadline and the window.
+// The rules for manipulating the field values are:
+// - The "ackDeadline" should be within a minimum and maximum value.
```

Comment:
These are still "checks" rather than operations, really. What I'm looking for is comments that mean when we're looking at a line of code, we can see whether the value is being used or computed in a valid way.

So it should be valid to say "compute the min of two deadlines" or "compute the max of two deadlines" (and ditto for windows and intervals", but it's not valid to say "compute the min of deadline X and window Y" or "assign this deadline value to an interval variable".

Later I'm going to try codify this in terms of internal-only structs - if I get it right, we should be able to see anywhere that we're violating those manipulation rules. It won't tell us whether the business rules about "must be at least X" are valid, but it'll tell us when the code is self-consistent.